Realm/Realm/Handles/SessionHandle.cs

```diff
@@ -267,57 +267,64 @@ public override void Unbind()
         [MonoPInvokeCallback(typeof(NativeMethods.SessionErrorCallback))]
         private static void HandleSessionError(IntPtr sessionHandlePtr, ErrorCode errorCode, PrimitiveValue message, IntPtr userInfoPairs, IntPtr userInfoPairsLength, bool isClientReset, IntPtr managedSyncConfigurationBaseHandle)
         {
-            try
+            var handle = new SessionHandle(null, sessionHandlePtr);
```

Comment:
Why did you remove the try/catch? User code could throw at line 284 or 288. Same as you've done at line 307, where you want to catch user exceptions that could come from line 311 or 316. And `handle` should be disposed also in the `finally` block of this new try/catch.

Reply:
I agree, I did it too hastily. I've tried to refactor it, so that all the user callbacks are executed at the end so we don't need to have two try/catch.